src/setup_mtc_place_from_pose.cpp

```diff
@@ -0,0 +1,227 @@
+      eliminated by failures in the stages before and after this one. **/
+  {
+    // Specify pose to generate for
+    auto stage = std::make_unique<moveit::task_constructor::stages::GeneratePose>("Generate pose");
+    stage->properties().configureInitFrom(moveit::task_constructor::Stage::PARENT);
+    stage->properties().set("marker_ns", "grasp_frame");
+    stage->setPose(place_pose.value());
+    stage->setMonitoredStage(task.value()->stages()->findChild("current state"));  // Hook into current state
+
+    // Compute IK
+    auto wrapper = std::make_unique<moveit::task_constructor::stages::ComputeIK>("Pose IK", std::move(stage));
+    wrapper->setMaxIKSolutions(kMaxIKSolutions);
+    wrapper->setTimeout(kIKTimeoutSeconds);
+    wrapper->setIKFrame(kHandFrameName);
+    wrapper->properties().configureInitFrom(moveit::task_constructor::Stage::PARENT, { "eef", "group" });
+    wrapper->properties().configureInitFrom(moveit::task_constructor::Stage::INTERFACE, { "target_pose" });
+    wrapper->setIgnoreCollisions(true);
+    container->add(std::move(wrapper));
+  }
+
+  /** Allow Collision
+      This stage allows collisions between the gripper and object for stages after this one (during the Close Hand,
+      Lift, and Retreat stages). */
+  {
+    auto stage =
+        std::make_unique<moveit::task_constructor::stages::ModifyPlanningScene>("Forbid collision (hand,object)");
+    stage->allowCollisions(kSceneObjectNameOctomap,
+                           task.value()
+                               ->getRobotModel()
+                               ->getJointModelGroup(kEndEffectorGroupName)
+                               ->getLinkModelNamesWithCollisionGeometry(),
+                           true);
+    container->add(std::move(stage));
+  }
+
+  /** Open Hand **/
+  {
+    auto stage =
+        std::make_unique<moveit::task_constructor::stages::MoveTo>("Open hand", mtc_joint_interpolation_planner);
+    stage->setGroup(kEndEffectorGroupName);
+    stage->setGoal(kHandOpenName);
+    container->add(std::move(stage));
+  }
+
+  /** Forbid Collision
+      This stage forbids collisions between the gripper and the object for subsequent stages. **/
+  {
+    auto stage =
+        std::make_unique<moveit::task_constructor::stages::ModifyPlanningScene>("Forbid collision (hand,object)");
+    stage->allowCollisions(kSceneObjectNameOctomap,
+                           task.value()
+                               ->getRobotModel()
+                               ->getJointModelGroup(kEndEffectorGroupName)
+                               ->getLinkModelNamesWithCollisionGeometry(),
+                           false);
+    container->add(std::move(stage));
+  }
+
+  /** Detach Object **/
+  {
+    auto stage = std::make_unique<moveit::task_constructor::stages::ModifyPlanningScene>("Detach");
+    stage->detachObject(kSceneObjectName, kHandFrameName);
+    container->add(std::move(stage));
+  }
```

Comment:
In the Pick From Pose behavior, we skipped attaching the object. Is this detach stage still necessary?